lib/fluent/plugin/out_elasticsearch.rb

```diff
@@ -156,18 +184,16 @@ def write(chunk)
         target_index = record.delete @target_index_key
       elsif @logstash_format
         if record.has_key?("@timestamp")
-          time = Time.parse record["@timestamp"]
+          dt = @time_parser.call(record["@timestamp"]) rescue Time.at(time).to_datetime
```

Comment:
The reason I went this way is that it was closer to the previous behaviour - `Time.parse` is really quite permissive and accepts many many formats, and I'd personally rather have the logs but with the wrong timestamp than drop them  and log an error on the aggregator which might not get seen.